pkg/backend/stack.go

```diff
@@ -34,6 +34,7 @@ import (
 // Stack is used to manage stacks of resources against a pluggable backend.
 type Stack interface {
 	Ref() StackReference                                    // this stack's identity.
+	Project() (*workspace.Project, error)                   // this stack's identity.
```

Comment:
I don't like this.

Reply:
Should project be exposed at such a high level? It seems to be an abstraction by the pulumi service on top of their stack naming conventions :woman_shrugging: 

I wonder what apply would look like if we were to remove workspace.Project from it.